Source/WebCore/platform/encryptedmedia/CDMProxy.cpp

```diff
@@ -332,6 +344,25 @@ bool CDMProxy::keyAvailable(const KeyIDType& keyID) const
     return keyAvailableUnlocked(keyID);
 }
 
+Ref<KeyHandleValuePromise> CDMProxy::getKeyHandleValue(const KeyIDType& keyID) const
+{
+    if (keyAvailable(keyID)) {
```

Comment:
We could get away with this structure, and only ever use the m_keyHandlePromises hashmap.
If the hashmap contained a resolved producer, the promise can be directly returned then.